src/utils/statsCheckings.js

```diff
@@ -0,0 +1,58 @@
+import Response from './ResponseHandler';
+/**
+ *
+ * @description view stats class
+ * @class StatsController
+ */
+export default class StatsController {
+  /**
+   * @param {object} allTrips
+   * @param {object} res
+   * @return {object} return stats
+   */
+  static async stats(allTrips, res) {
+    const createdTrips = allTrips.length;
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+    let lastMonth, openRequests, approvedRequests, rejectedRequets, previousMonths, oldTrips;
+    const newArry = [], previousMonthsArray = [],
+      previousYearsArray = [], openRequestArr = [],
+      approvedRequestArr = [], rejectedRequetsArr = [];
+    allTrips.forEach(element => {
+      switch (element.status) {
+        case 'open':
+          openRequestArr.push(element);
+          openRequests = openRequestArr.length;
+          break;
+        case 'approved':
+          approvedRequestArr.push(element);
+          approvedRequests = approvedRequestArr.length;
+          break;
+        case 'rejected':
+          rejectedRequetsArr.push(element);
+          rejectedRequets = rejectedRequetsArr.length;
+          break;
+        default:
+      }
+      if (element.createdAt.getFullYear() === currentYear) {
+        if (element.createdAt.getMonth() < (currentMonth - 1)) {
+          newArry.push(element);
+          lastMonth = newArry.length;
+        } else {
+          previousMonthsArray.push(element);
+          previousMonths = previousMonthsArray.length;
+        }
```

Comment:
the else statement seems like it will find trips made in the upcoming months rather than the previous one

Reply:
sure, that's correct! I am going to rename it. Thanks

Reply:
Actually I am giving stats based on the time the request was created.